.github/workflows/deploy.testnet.yml

```diff
@@ -0,0 +1,186 @@
+---
+name: TestNet Workflow
+on:  # yamllint disable-line rule:truthy
+  workflow_dispatch:
+  push:
+    branches:
+      - 'release/**'
+
+permissions:
+  id-token: write
+  contents: read
+  security-events: write
+
+jobs:
+  gosec:
+    name: Gosec and Publish
+    uses: 0xpolygon/polygon-edge/.github/workflows/security.yml@develop
+
+  build:
+    uses: 0xpolygon/polygon-edge/.github/workflows/build.yml@develop
+    name: Build
+
+  test:
+    uses: 0xpolygon/polygon-edge/.github/workflows/test.yml@develop
+    name: Test
+    needs: build
+
+  deploy_testnet:
+    name: Update TestNet
+    needs: build
+    concurrency: deploy_testnet
+    environment:
+      name: testnet
+      url: https://rpc.us-east-1.tn.testing.psdk.io/
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download Polygon Edge Artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: polygon-edge
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      - name: Notify Slack
+        uses: slackapi/slack-github-action@v1.22.0
+        continue-on-error: true
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_DEPLOYMENT_WEBHOOK_URL }}
+          SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+        with:
+          payload: |
+            {
+              "blocks": [
+                {
+                  "type": "header",
+                  "text": {
+                    "type": "plain_text",
+                    "text": "Testnet Deployment - Started"
+                  }
+                },
+                {
+                  "type": "section",
+                  "text": {
+                    "type": "mrkdwn",
+                    "text": "Testnet Deployment Status: ${{ job.status }}\n<${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|Job Status>"
+                  }
+                }
+              ]
+            }
+
+      - name: Publish Binary package to S3
+        run: |
+          aws s3 cp ./polygon-edge.tar.gz s3://${{ secrets.POLYGON_EDGE_ARTIFACT_BUCKET }} --metadata "{\"commit_sha\":\"${GITHUB_SHA}\"}"
+      - name: Deploy to Testnet
+        continue-on-error: true
+        env:
+          VALIDATOR_ASGS: ${{ secrets.VALIDATOR_ASGS }}
+        run: |
+          echo "Deploying new TestNet..."
```

Comment:
Are we going to reset the data between deployments in testnets?

Reply:
Not for Testnet. The Testnet deployment currently initiates instance refreshes on each ASG **sequentially**. On init via user-data, the new EC2s attach their respective EBS with the chain data persistent.

DevNet deployment however runs an SSM document that stops the service and cleans up the data directory prior to initiating instance refresh **asynchronously**

An alternative Testnet deployment is running SSM to handle the deployment scope on the instance without a re-cycle.

Reply:
LGTM